src/main/java/com/adaptris/core/mail/MailProducer.java

```diff
@@ -192,13 +192,20 @@ public String getSubject() {
     return subject;
   }
 
+  /**
+   * @deprecated since 3.10.0, slated for removal in 3.11.0, use message resolver instead.
+   */
+  @Deprecated
```

Comment:
@albinoloverats Could you also add a note into the main javadocs (that details all the hard-coded metadata keys) about how they're deprecated and will be removed w/o warning...
-> Also deprecate the associated EmailConstants# value... (with an Removal annotation as well) -> then it makes it easy for us to do a "search" for `Removal(version="3.11.0")` when we actually get round to removing it.

Reply:
@quotidian-ennui does that mean we will need to start making changes to the DefaultMailConsumer to use data output parameters? As it currently uses many of the constants to add metadata.

Reply:
What I mean is that "if someone configures the metadata key" `emailcc` (or whatever it is) with some data, then the producer will always use that regardless of what they have configured in the `cc-list`; it's silent behaviour, that's documented but not explicitly configured.

So; we want to _force them to use_ `%message{emailcc}` explicitly in their configuration...

So, if we deprecate the EmailConstants#emailcc then hopefully someone will (eventually) get the idea that they shouldn't be just silently adding it and hoping for the best, then need to explicitly change their config so `%messag{emailcc}` is explicit in the config.

Also by mentioning that in the various javadoc things that mention those constants by name.

Reply:
Don't forgot to add our custom @Removal annotation. (It should be obvious-ish).